### PR TITLE
[LL-6803] generalized sign transaction flow

### PR DESCRIPTION
**Pull request by ghost**
## Context (issues, jira)

https://ledgerhq.atlassian.net/browse/LL-6803

⚠️ linked PR:
~🔗  https://github.com/LedgerHQ/ledger-live-common/pull/1328~
🔗  https://github.com/LedgerHQ/ledger-live-desktop/pull/4112
~🔗  https://github.com/LedgerHQ/ledger-live-platform-sdk/pull/18~

## Description / Usage

Add transactions adapters from LLP-SDK tx to LLC tx, used by generalisation work done in the [desktop PR](https://github.com/LedgerHQ/ledger-live-desktop/pull/4112)

Also update the `useFeesStrategy` for BTC and ETH to drop invariant and return empty array `[]` when no network info available yet (needed by [BTC](https://github.com/LedgerHQ/ledger-live-desktop/blob/develop/src/renderer/families/bitcoin/SendAmountFields.js#L60) and [ETH](https://github.com/LedgerHQ/ledger-live-desktop/blob/develop/src/renderer/families/ethereum/SendAmountFields.js#L33) `SendAmountFields` components)

## Expectations

- [x] **Test coverage: The changes of this PR are covered by test.** Unit test were added with mocks when depending on a backend/device.
- [x] **No impact: The changes of this PR have ZERO impact on the userland.** Meaning, we can use these changes without modifying LLD/LLM at all. It will be a "noop" and the maintainers will be able to bump it without changing anything.

Should have no impact on userland since it mainly adds functions. Must be merged before the related LLD PR since LLD relies on the newly implemented functions in this PR.

----

## TODO

- [x] Add tests regarding platform converter / adapters
- [x] publish new version of SDK and update dependency (to fix Polkadot transaction mode type error)

### CODE REVIEW

File: src/families/polkadot/platformAdapter.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import { PolkadotOperationMode, Transaction } from "./types";
+
+export const CAN_EDIT_FEES = false;
+
+export const convertToLiveTransaction = (
+  tx: PlatformTransaction
+): Partial<Transaction> => ({
+  ...tx,
+  era: `${tx.era}`,
+  mode: tx.mode as PolkadotOperationMode, // FIXME: update SDK polkadot transaction type to include a PolkadotOperationMode type
```
Comment (achabrolin-ledger) [inline src/families/polkadot/platformAdapter.ts:11]: Should be fixed by https://github.com/LedgerHQ/live-app-sdk/pull/22

**Review by lkrust — CHANGES_REQUESTED** [review]
-

**Review by lkrust — APPROVED** [review]

File: src/platform/converters.ts [hunk 2]
```diff
@@ -29,3 +41,66 @@ export function currencyToPlatformCurrency(
     })),
   };
 }
+
+export const getPlatformTransactionSignFlowInfos = (
```
Comment (juan-cortes) [inline src/platform/converters.ts:45]: Isn't this the ideal place for something that gets exported `byFamily` and we just provide the family to the object to get the method? Like we do with any of the generated code by family, ie https://github.com/LedgerHQ/ledger-live-common/blob/master/src/generated/hw-getAddress.ts you would then import this byFamily and pas the platform 🤷🏼‍♂️

**Review by juan-cortes — APPROVED** [review]
lgtm

File: src/platform/converters.ts (hunk 2 above)
Comment (achabrolin-ledger) [reply src/platform/converters.ts:45]: Maybe, I'm not yet familiar with this `byFamily` kind of abstraction. I'll take a look at this and propose an update accordingly

Comment (achabrolin-ledger) [reply src/platform/converters.ts:40]: Updated here e0c3b5ffeaea9e200421fab8bc1afb075b971dcf

**Review by IAmMorrow — APPROVED** [review]

File: src/families/bitcoin/react.ts [hunk 3]
```diff
@@ -1,9 +1,10 @@
-import invariant from "invariant";
 import type { NetworkInfo, Transaction } from "./types";
 import type { FeeStrategy, Account } from "../../types";
 export const useFeesStrategy = (a: Account, t: Transaction): FeeStrategy[] => {
   const networkInfo = t.networkInfo;
-  invariant(networkInfo, "no network info");
+
+  if (!networkInfo) return [];
```
Comment (valpinkman) [inline src/families/bitcoin/react.ts:6]: this actually changes the behavior, wouldn't it be better with a 
```suggestion
  if (!networkInfo) throw new Error("no network info")
```

?

**Review by valpinkman — APPROVED** [review]

**Pull request merged**